jni/cmake/init-faiss.cmake

```diff
@@ -111,10 +111,10 @@ endif()
 if(NOT DEFINED AVX512_SPR_ENABLED)
     # Check if the system is Intel(R) Sapphire Rapids or a newer-generation processor
     execute_process(COMMAND bash -c "lscpu | grep -q 'GenuineIntel' && lscpu | grep -i 'avx512_fp16' | grep -i 'avx512_bf16' | grep -i 'avx512_vpopcntdq'" OUTPUT_VARIABLE SPR_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
-    if (AND NOT "${SPR_FLAGS}" STREQUAL "")
```

Comment:
Good catch @sam-herman . @mulugetam Can you check this and make sure this change makes sense?

Reply:
It is a valid change. I fixed it in 2.x and 2.19 but somehow missed backporting it to main